client/app/bundles/course/assessment/submission/containers/TestCaseView.jsx

```diff
@@ -121,15 +121,15 @@ class VisibleTestCaseView extends Component {
     );
   }
 
-  static renderOutputStream(outputStreamType, output) {
+  static renderOutputStream(outputStreamType, output, showStaffOnlyWarning) {
     return (
       <Card>
         <CardHeader
           showExpandableButton
           title={
             <React.Fragment>
               <FormattedMessage {...translations[outputStreamType]} />
-              {VisibleTestCaseView.renderStaffOnlyOutputStreamWarning()}
+              {showStaffOnlyWarning ? VisibleTestCaseView.renderStaffOnlyOutputStreamWarning() : null}
```

Comment:
You can use a slightly different pattern of defining this, can read the [link here](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator). Something like `{ boolean && renderSomething }`.